ddtrace.gemspec

```diff
@@ -29,17 +29,20 @@ Gem::Specification.new do |spec|
     raise 'RubyGems 2.0 or newer is required to protect against public gem pushes.'
   end
 
-  # rubocop:disable all
-  # DEV: `spec.files` is more problematic than a simple Rubocop pass.
   spec.files =
-    `git ls-files -z`
```

Comment:
This change is due to the new [Packaging/GemspecGit](https://docs.rubocop.org/rubocop-packaging/cops_packaging.html#packaginggemspecgit) cop.

It's better practice to not depend on a locally installed `git` to build the gem. This can help downstream maintainers distributed our gem.

Reply:
I have... mixed feelings about this change.

On one hand, I do agree that depending on git here is icky, and it can lead to wonky behaviors in cases where git is not installed or doesn't have access to the `.git` folder. (This has happened to me, since I do weird things)

But I see two downsides here:
* 1️⃣  For the folders where we're using wildcards (*), we run the risk of including unexpected files in the release. "Oops I had some leftover temp files in my folder"
* 2️⃣  For the folders where we're not using wildcards, we run the risk of forgetting to include files that were expected to be in the release. And right now our CI runs with the full tree so we wouldn't even notice, if for instance we added `lib2/` and then forgot to update it here.

Even looking at the list this PR adds and comparing it with the files currently shipped I find some gaps. Maybe they were deliberate?

```
.editorconfig # <-- not going to be shipped
.gitignore # <-- not going to be shipped
CHANGELOG.md
CONTRIBUTING.md # <-- not going to be shipped
LICENSE
LICENSE-3rdparty.csv
LICENSE.Apache
LICENSE.BSD3
NOTICE
README.md
bin/ddtracerb
ddtrace.gemspec # <-- not going to be shipped (I think...?)
docs/0.x-trace.png # <-- not going to be shipped
docs/1.0-trace.png # <-- not going to be shipped
docs/AutoInstrumentation.md
docs/Deprecation.md
docs/DevelopmentGuide.md
docs/GettingStarted.md
docs/ProfilingDevelopment.md
docs/PublicApi.md
docs/UpgradeGuide.md
ext/...
lib/...
```

Having said all that, I think the issues are solvable without discarding this approach.

I'm thinking we can use two things to our advantage:
a) CI
b) Extra checks while running `rake build` to package a release

For instance, to solve issue 2️⃣ of "missing files" that should've been included we could have a rake task that runs during CI to compare the list in `spec.files` to the actual output of `git ls-files -z` + filters we had before. The lists should match; if they don't, something is off.

For the issue of including extra files in the release (1️⃣) we could run the same/a similar task before packaging, which would detect if "oops you have a leftover file in your development tree".

Something like that?

Reply:
Sorry for not posting this information, but these omissions are deliberate:
```
.editorconfig # <-- not going to be shipped
.gitignore # <-- not going to be shipped
CHANGELOG.md
CONTRIBUTING.md # <-- not going to be shipped
LICENSE
LICENSE-3rdparty.csv
LICENSE.Apache
LICENSE.BSD3
NOTICE
README.md
bin/ddtracerb
ddtrace.gemspec # <-- not going to be shipped (I think...?)
docs/0.x-trace.png # <-- not going to be shipped
docs/1.0-trace.png # <-- not going to be shipped
```

I was inspecting how some popular gems are packaged, and the ones that don't use `git` seem very simple: [activesupport](https://github.com/rails/rails/blob/main/activesupport/activesupport.gemspec#L20), [resque](https://github.com/resque/resque/blob/e5e03962c297bbc9392e17bfccb67799d17c7778/resque.gemspec#L12-L16), [rack](https://github.com/rack/rack/blob/59c50ddf8998dbd2a6b472525bc1280da71f6c7b/rack.gemspec#L20-L21).

Reply:
I thought about creating a test for it, but thought it was overkill.

I'm ok with it.

Reply:
Tests are here, @ivoanjo!